doc/manual/command-ref/conf-file.xml

```diff
@@ -240,6 +242,70 @@ false</literal>.</para>
 
   </varlistentry>
 
+  <varlistentry xml:id="conf-diff-hook"><term><literal>diff-hook</literal></term>
+  <listitem>
+    <para>
+      Absolute path to an executable capable of diffing build results.
+      The hook executes if <xref linkend="conf-run-diff-hook" /> is
+      true, and the output of a build is known to not be the same.
+      This program is not executed to determine if two results are the
+      same.
+    </para>
+
+    <para>
+      The diff hook is executed by the same user and group who ran the
+      build. However, the diff hook does not have write access to the
+      store path just built.
+    </para>
+
+    <para>The diff hook program receives three parameters:</para>
+
+    <orderedlist>
+      <listitem>
+        <para>
+          A path to the previous build's results
+        </para>
+      </listitem>
+
+      <listitem>
+        <para>
+          A path to the current build's results
+        </para>
+      </listitem>
+
+      <listitem>
+        <para>
+          The path to the build's derivation
+        </para>
+      </listitem>
+
+      <listitem>
+        <para>
+          The path to the build's scratch directory. This directory
+          will exist only if the build was run with
+          <option>--keep-failed</option>.
+        </para>
+      </listitem>
+    </orderedlist>
+
+    <para>The diff hook should not print data to stderr or stdout, as
```

Comment:
What exactly do you mean here? I guess it _should_ print to `stderr` right? Otherwise how does the user figure out the output of the diff hook?

Reply:
No. None of the output printed by the diff hook is shown to the user, it all goes to the nix-daemon log.

Reply:
But that doesn't mean it is 'bad' for a diff-hook to output something right?  The sentence now seems to imply that diff hooks shouldn't print results. But we do want them to print results, such that we can actually see what the diff-hook found out, when looking at the nix-daemon logs.

Reply:
Sure, though writing the diff report elsewhere might be a better bet. Do you have a suggestion on exactly how to reword this?

Reply:
> Note that if the diff hook prints data to stderr or stdout, and <command>nix-daemon</command> is enabled,  this is not displayed directly to the user. Instead,
it will be printed in the <command>nix-daemon</command> log

Reply:
I don't remember how this is implemented exactly, but I think it might be pretty easy to forward the output of the hook to the client. Similar to how this is done for build logs.